src/util/battery/batterylinux.h

```diff
@@ -3,13 +3,18 @@
 
 #include "util/battery/battery.h"
 
+// Contains 'signals' so we undefine Qt's 'signals'.
+#undef signals
```

Comment:
This affects all code that includes this header file - not acceptable. Instead use forward declarations and only undef in the .cpp file.

Reply:
UpClient is a typedef:
```
typedef struct {
         GObject                 parent;
         UpClientPrivate        *priv;
} UpClient;
```
How can this be forward declared? Any help would be appreciated ;)

Reply:
The `typedef` keyword shouldn't matter, it is just a struct. For a solution that is independent of how `UpClient` is defined use a `void*` pointer in the class definition and `static_cast` in the .cpp file.

Reply:
I didn't manage to do a forward declaration because the struct is unnamed. 
So I went for the `void*/static_cast` solution.